### PR TITLE
Singing Voice Synthesis Task for ESPnet

### PR DESCRIPTION
**Pull request by ftshijt**
This is the singing voice synthesis task for ESPnet, as noted in https://github.com/espnet/espnet/issues/4437

For now, we have supported a sample recipe and the corresponding setups (including RNN and RNN-DP models). 

Some major updates from original Muskits:
- We replace the raw music score from MIDI to MusicXML for easier usage in various cases
- We fixed some legacy issues with Muskits

It would be great if you could have some suggestions and comments.

We intend to distribute the work after this initial one. Therefore, we would probably add test functions in the following PR.

### CODE REVIEW

File: espnet2/train/preprocessor.py [hunk 1]
```diff
@@ -944,3 +946,236 @@ def _speech_process(
...
+            midis = []
+            xml_timeseq = []
+            phn_cnt = 0
+            vowel = ["a", "i", "u", "e", "o"]
+            for i in range(len(syllables)):
+                # NOTE: Some phonemes are tagged differently
+                if syllables[i] == "へ":
+                    phn = ["h", "e"]
+                else:
+                    phn = pyopenjtalk.g2p(syllables[i]).split(" ")
```
Comment (ftshijt) [inline espnet2/train/preprocessor.py:1051]: can we instead use `self.tokenizer` to do this part? The current procedure breaks the CI at https://github.com/espnet/espnet/actions/runs/3141052108/jobs/5103071721 . @A-Quarter-Mile

File: espnet2/bin/svs_inference.py [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+from espnet2.torch_utils.device_funcs import to_device
+from espnet2.torch_utils.set_all_random_seed import set_all_random_seed
+
+from espnet2.svs.naive_rnn.naive_rnn import NaiveRNN
+from espnet2.svs.naive_rnn.naive_rnn_dp import NaiveRNNDP
+
+# from espnet2.svs.glu_transformer.glu_transformer import GLU_Transformer
+# from espnet2.svs.xiaoice.XiaoiceSing import XiaoiceSing
+# from espnet2.svs.xiaoice.XiaoiceSing import XiaoiceSing_noDP
+
```
Comment (sw005320) [inline espnet2/bin/svs_inference.py:31]: Maybe, we can remove these lines?

**Review by sw005320 — APPROVED** [review]
It looks good to me.
Since it is separated from the others at the task level, I don't think we would have some critical issues at the design level.

My general suggestion is to add a test (which is already mentioned that it will be done in the following PR) and document (e.g., main REAME.md and other places).

File: espnet2/bin/svs_inference.py (hunk 2 above)
Comment (ftshijt) [reply espnet2/bin/svs_inference.py:31]: Sure. These are models to be added in future PRs. Will remove it for now

Comment (sw005320) [reply espnet2/bin/svs_inference.py:31]: If there are some intentions, you can keep these lines.

**Comment by ftshijt** [comment]
The CI seems to be failed due to some `flack8` issues

**Comment by ftshijt** [comment]
Please let us know if the PR looks good~ It would be great to merge for further distribution of works.

**Comment by ftshijt** [comment]
As this is quite a dependency for following works, I would want to merge it for now~ There are plenty of TODOs to make it better. Please feel free to provide any comments if you have any~

**Pull request merged**